services/121-service/src/debit-cards-intersolve-visa/debit-cards-intersolve-visa.service.ts

```diff
@@ -324,10 +267,132 @@ export class DebitCardsIntersolveVisaService {
     referenceId: string,
     programId: number,
   ): Promise<void> {
-    const registration = await this.getRegistrationOrThrow({
-      referenceId,
-      programId,
+    const registration =
+      await this.registrationUtilsService.getRegistrationOrThrow({
+        referenceId,
+        programId,
+      });
+    const contactInfo: DebitCardsContactInfo =
+      await this.getContactInformation(registration);
+    await this.sendCustomerInformationToIntersolve({
+      registration,
+      contactInfo,
+    });
+  }
+
+  public async sendCustomerInformationToIntersolve({
+    registration,
+    contactInfo,
+  }: {
+    registration: RegistrationEntity;
+    contactInfo: DebitCardsContactInfo;
+  }): Promise<void> {
+    const registrationHasVisaCustomer =
+      await this.intersolveVisaService.hasIntersolveCustomer(registration.id);
+
+    if (registrationHasVisaCustomer) {
+      await this.intersolveVisaService.sendUpdatedCustomerInformation({
+        registrationId: registration.id,
+        ...contactInfo,
+      });
+    }
+  }
+
+  public async getContactInformation(
+    registration: RegistrationEntity,
+  ): Promise<DebitCardsContactInfo> {
+    const fieldNames: DebitCardsContactInfoKeys[] = [
+      FspAttributes.addressStreet,
+      FspAttributes.addressHouseNumber,
+      FspAttributes.addressHouseNumberAddition,
+      FspAttributes.addressPostalCode,
+      FspAttributes.addressCity,
+      FspAttributes.phoneNumber,
+      FspAttributes.fullName,
+    ];
+
+    const registrationData =
+      await this.registrationDataScopedRepository.getRegistrationDataArrayByName(
+        registration,
+        fieldNames,
+      );
+
+    if (!registrationData || registrationData.length === 0) {
+      throw new HttpException(
+        `No registration data found for referenceId: ${registration.referenceId}`,
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    const mappedRegistrationData = registrationData.reduce(
+      (acc, { name, value }) => {
+        acc[name] = value;
+        return acc;
+      },
+      {},
+    );
+
```

Comment:
I would use something like this to get the required intersolve visa data from the registration:
```
  public async getContactInformation(
    referenceId: string,
  ): Promise<DebitCardsContactInfo> {
   	 const intersolveVisaRegistrationDataFields = FSP_SETTINGS[
      Fsps.intersolveVisa
    ].attributes.map((attribute) => attribute.name);

    const registration = (
      await this.registrationPaginationService.getRegistrationViewsByReferenceIds(
        {
          referenceIds: [referenceId],
          programId,
          select: intersolveVisaRegistrationDataFields,
        },
      )
    )[0];
```